CancerAidDev/elm-form-builder
# [PT-2694] Add missing update functionality for HttpSearchableSelectField

- add field to example to test out that it is working properly
- add missing `httpSearchableSelect` constructor
- missed a few things - should be fixed now
  - add `HttpSearchableSelectField` to `genericUpdateStringField` and `updateShowDropdown` - allows the dropdown to be opened and closed
  - add `HttpSearchableSelectField` to `updateSearchableSelectInput`
  - add `HttpSearchableSelectField` to `updateRemoteOptions`
  - add `HttpSearchableSelectField` to `getUrl`
  - add `updateHiddenByFields` to reset the hidden field value when it's `hiddenBy` field is updated to an empty/false value
- added a reset button so that searchable select fields that are optional (`required` is not set to `Yes`) can be reset (currently if you select a value, I don't think there's a way to remove it and have it empty)
